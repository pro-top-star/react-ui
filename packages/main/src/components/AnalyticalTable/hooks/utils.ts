export const makeTemplateColumns = (columns, resizedColumns) => {
  return columns
    .map((column) => {
      if (resizedColumns[column.id]) {
        return `${resizedColumns[column.id]}px`;
      }
      return `minmax(${column.minWidth}px, ${column.width || '1fr'})`;
    })
    .join(' ');
};
