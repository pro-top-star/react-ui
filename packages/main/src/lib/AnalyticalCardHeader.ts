import { AnalyticalCardHeader } from '../components/AnalyticalCard/header/AnalyticalCardHeader';

export { AnalyticalCardHeader };
