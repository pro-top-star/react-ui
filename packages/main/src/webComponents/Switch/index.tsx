import { Event } from '@ui5/webcomponents-react-base/lib/Event';
import { withWebComponent } from '@ui5/webcomponents-react/lib/withWebComponent';
import UI5Switch from '@ui5/webcomponents/dist/Switch';
import React, { FC } from 'react';
import { WithWebComponentPropTypes } from '../../internal/withWebComponent';

export interface SwitchPropTypes extends WithWebComponentPropTypes {
  checked?: boolean; // @generated
  disabled?: boolean; // @generated
  textOn?: string; // @generated
  textOff?: string; // @generated
  graphical?: boolean; // @generated
  onChange?: (event: Event) => void; // @generated
}

/**
 * <code>import { Switch } from '@ui5/webcomponents-react/lib/Switch';</code>
 * <br />
 * <a href="https://sap.github.io/ui5-webcomponents/playground/components/Switch" target="_blank">UI5 Web Components Playground</a>
 */
const Switch: FC<SwitchPropTypes> = withWebComponent<SwitchPropTypes>(UI5Switch);

Switch.displayName = 'Switch';

export { Switch };
