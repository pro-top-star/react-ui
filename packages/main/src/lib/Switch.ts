import { Switch } from '../webComponents/Switch';
import type { SwitchPropTypes } from '../webComponents/Switch';

export { Switch };
export type { SwitchPropTypes };
