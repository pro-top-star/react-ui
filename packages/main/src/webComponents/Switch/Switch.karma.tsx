import React from 'react';
import { expect, use } from 'chai';
import { matchSnapshot } from 'chai-karma-snapshot';
import { Switch } from '../../lib/Switch';
import { mountThemedComponent } from '@shared/tests/utils';

use(matchSnapshot);

describe('Switch', () => {
  it('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Switch />);
    expect(wrapper.debug()).to.matchSnapshot();
  });
});
