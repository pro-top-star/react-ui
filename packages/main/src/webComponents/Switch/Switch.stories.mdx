import { Meta, Story, Canvas, ArgsTable } from '@storybook/addon-docs/blocks';
import { Switch } from '@ui5/webcomponents-react/lib/Switch';
import { CSSProperties, Ref } from 'react';
import { createSelectArgTypes } from '@shared/stories/createSelectArgTypes';
import { DocsHeader } from '@shared/stories/DocsHeader';
import { DocsCommonProps } from '@shared/stories/DocsCommonProps';

<Meta
  title="UI5 Web Components / Switch"
  component={Switch}
  argTypes={{
    ...createSelectArgTypes({}),
    ...DocsCommonProps
  }}
  args={{
    style: {},
    className: '',
    tooltip: '',
    slot: '',
    ref: null
  }}
/>

<DocsHeader />

<br />

## Example

<Canvas>
  <Story name="Default">
    {(args) => {
      return <Switch {...args} />;
    }}
  </Story>
</Canvas>

## Properties

<ArgsTable story="." />
<div style={{ fontFamily: 'var(--sapFontFamily)', fontSize: 'var(--sapFontSize)', color: 'var(--sapTextColor)' }}>
  <h3>Keyboard Handling</h3>
  The state can be changed by pressing the Space and Enter keys.
</div>
