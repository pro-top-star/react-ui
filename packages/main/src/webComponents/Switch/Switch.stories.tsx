import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { Switch } from '@ui5/webcomponents-react/lib/Switch';
import React from 'react';

export default {
  title: 'UI5 Web Components / Switch',
  component: Switch
};

export const generatedDefaultStory = () => (
  <Switch
    checked={boolean('checked', false)}
    disabled={boolean('disabled', false)}
    graphical={boolean('graphical', false)}
    textOff={text('textOff', '')}
    textOn={text('textOn', '')}
    onChange={action('onChange')}
  />
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
