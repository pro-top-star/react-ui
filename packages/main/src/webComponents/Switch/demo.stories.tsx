import { action } from '@storybook/addon-actions';
import { boolean, text } from '@storybook/addon-knobs';
import { Switch } from '@ui5/webcomponents-react/lib/Switch';
import React from 'react';

export default {
  title: 'UI5 Web Components / Switch',
  component: Switch
};

export const generatedDefaultStory = () => (
  <Switch
    checked={boolean('checked', false)}
    disabled={boolean('disabled', false)}
    textOn={text('textOn', 'Yes')}
    textOff={text('textOff', 'No')}
    graphical={boolean('graphical', false)}
    onChange={action('onChange')}
  />
);

generatedDefaultStory.story = {
  name: 'Generated default story'
};
