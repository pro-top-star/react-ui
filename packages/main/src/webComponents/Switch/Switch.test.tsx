import { mountThemedComponent } from '@shared/tests/utils';
import { Switch } from '@ui5/webcomponents-react/lib/Switch';
import React from 'react';

describe('Switch', () => {
  test('Basic Test (generated)', () => {
    const wrapper = mountThemedComponent(<Switch />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
