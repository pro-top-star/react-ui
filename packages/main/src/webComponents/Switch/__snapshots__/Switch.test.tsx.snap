// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch Basic Test (generated) 1`] = `<ui5-switch />`;
